File: Src/WitsmlExplorer.Frontend/models/dataWorkOrder/dataSourceConfigurationSet.ts
import DataSourceConfiguration from "models/dataWorkOrder/dataSourceConfiguration";

export default interface DataSourceConfigurationSet {
  uid: string;
  dataSourceConfigurations: DataSourceConfiguration[];
}

export const getLastDataSourceConfiguration = (
  dataSourceConfigurationSet: DataSourceConfigurationSet
) =>
  dataSourceConfigurationSet?.dataSourceConfigurations?.reduce(
    (last, current) =>
      last == null || (current.versionNumber ?? 0) > (last.versionNumber ?? 0)
        ? current
        : last,
    null
  );

File: Src/WitsmlExplorer.Api/Models/DataWorkOrder/DataSourceConfigurationSet.cs
using System.Collections.Generic;
using System.Linq;

using Witsml.Data.DataWorkOrder;

namespace WitsmlExplorer.Api.Models.DataWorkOrder;

public class DataSourceConfigurationSet
{
    public string Uid { get; set; }

    public List<DataSourceConfiguration> DataSourceConfigurations { get; set; }

}

public static class DataSourceConfigurationSetExtensions
{
    public static WitsmlDataSourceConfigurationSet ToWitsml(this DataSourceConfigurationSet configurationSet)
    {
        return new WitsmlDataSourceConfigurationSet
        {
            Uid = configurationSet.Uid,
            DataSourceConfigurations = configurationSet.DataSourceConfigurations.Select(configuration => configuration?.ToWitsml())?.ToList(),
        };
    }
}
